libs/agno/agno/tools/openai.py

```diff
@@ -36,3 +66,119 @@ def transcribe_audio(self, audio_path: str) -> str:
         except Exception as e:
             logger.error(f"Failed to transcribe audio: {str(e)}")
             return f"Failed to transcribe audio: {str(e)}"
+
+    def generate_image(
+        self,
+        agent: Agent,
+        prompt: str,
+        model_name: Literal["dall-e-2", "dall-e-3"] = "dall-e-3",
```

Comment:
How would the agent know to pick a model. I would suggest the model is set on the init?

Also change `n` to `number_of_images`